CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing to `tracing-active-tree`
+
+Thanks for your interest in contributing to `tracing-active-tree`! This document outlines some of the conventions on building, running, and testing `tracing-active-tree`, the development workflow, commit message formatting, contact points and other resources.
+
+## Building and setting up a development workspace
+
+`tracing-active-tree` is mostly written in Rust, We are currently using the Rust nightly toolchain.
+
+### Prerequisites
+
+To build `tracing-active-tree` you'll need to at least have the following installed:
+
+* `git` - Version control
+* [`rustup`](https://rustup.rs/) - Rust installer and toolchain manager
+* `make` - Build tool (run common workflows)
+
+### Getting the repository
+
+```bash
+git clone https://github.com/tikv/tracing-active-tree.git
+cd tracing-active-tree
+```
+
+### Configuring your Rust toolchain
+
+`rustup` is the official toolchain manager for Rust, similar to `rvm` or `rbenv` from the Ruby world.
+
+`tracing-active-tree` is pinned to a version of Rust using a `rust-toolchain` file. `rustup` and `cargo` will automatically use this file. We also use the `rustfmt` and `clippy` components, to install those:
+
+```bash
+rustup component add rustfmt
+rustup component add clippy
+```
+
+### Building and testing
+
+During interactive development, you may prefer using `cargo check`, which will parse, borrow check, and lint your code, but not actually compile it:
+
+```bash
+cargo check --all
+```
+
+`nextest` is necessary for `make test` & `make unit-test` to work. Install the crate with:
+
+```bash
+cargo install cargo-nextest --locked
+```
+
+You can run the test suite alone, or just run a specific test:
+
+```bash
+# Run the full suite
+make test
+```
+
+`tracing-active-tree` follows the Rust community coding style. We use Rustfmt and [Clippy](https://github.com/Manishearth/rust-clippy) to automatically format and lint our code. Using these tools is checked in our CI. These are as part of `make check`.
+
+```bash
+make check
+```
+
+### Finding something to work on
+
+For beginners, we have prepared many suitable tasks for you. Checkout our [Help Wanted issues](https://github.com/tikv/tracing-active-tree/issues?q=is%3Aopen+is%3Aissue+label%3Astatus%2Fhelp-wanted) for a list, in which we have also marked the difficulty level.
+
+If you are planning something big, for example, relates to multiple components or changes current behaviors, make sure to open an issue to discuss with us before going on.
+
+### Linking issues
+
+Code repositories in `tracing-active-tree` community require **ALL** the pull requests referring to its corresponding issues. In the pull request body, there **MUST** be one line starting with `Issue Number:` and linking the relevant issues via the [keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), for example:
```

Comment:
I think it is a little verbose by now, especially there are only three of us. Maybe we can provide a PR template which guides to link an issue, but it's not a mandatory demand.

Reply:
removed.